nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/record/script/AbstractScriptedRecordFactory.java

```diff
@@ -42,9 +42,13 @@ public void setup() {
 
         if (scriptNeedsReload.get() || recordFactory.get() == null) {
             if (ScriptingComponentHelper.isFile(scriptingComponentHelper.getScriptPath())) {
-                reloadScriptFile(scriptingComponentHelper.getScriptPath());
+                if (!reloadScriptFile(scriptingComponentHelper.getScriptPath())) {
```

Comment:
I'm not sure we need to raise an exception here since reloadScriptFile and reloadScriptBody already add a validation failure and logs it.

Reply:
The exception was supposed to prevent the CS from enabling, Andy pointed out the same issue. I thought I'd tested that but apparently did not, will take a closer look

Reply:
I think we need the exception to indicate to the framework that the CS was not enabled properly, even though the processor is still allowed to run and the validation errors are already logged (hopefully only once now)